src/css.js

```diff
@@ -303,7 +316,8 @@ jQuery.extend( {
 			num = parseFloat( val );
 			return extra === true || isFinite( num ) ? num || 0 : val;
 		}
-		return val;
+
+		return val === undefined ? undefined : jQuery.trim( val );
```

Comment:
To clarify, by a proper test for that I meant dynamically creating a stylesheet with contents like:
```css
.test-custom-props {
    --prop1: val1;
    --prop2:val2;
    --prop3:val3 ;
    --prop4: "val4" ;
}
```
and make sure all spaces/quotes/etc. are there in the computed values (I hope browsers are not buggy here!).